src/js/components/Drop/DropContainer.js

```diff
@@ -292,10 +293,8 @@ const DropContainer = forwardRef(
         }
       };
     }, [
-      align,
```

Comment:
I think removing these from the dependency array will cause other defects since the code inside will continue to use old values if either of those dependencies actually change. 

I'm guessing that really the  caller shouldn't be passing in objects that are constructed again each render or this very large useEffect function needs to be refactored so it doesn't have to depend on those props.

Reply:
@MikeKingdom agreed about not removing from dependency array, i just asked @taysea about that and we were looking at how to memoize the props at the component level. i was looking at useMemo for the align prop, but still unsure if this makes the most sense since it relies on an empty dependency array

would refactoring require abstracting the logic depending on the align prop from that useEffect?

Reply:
The short answer is it is too late to memoize within here if it's coming from a prop. If the prop coming in is always a different instance on each render, your memoize will re-memorize it for the same reason it was causing the useEffect to get re-triggered, even when the contains of the object in the prop are identical to before. Either the caller has to memoize it or the code here has to be refactored so that the useEffect doesn't have a copy of it. In practice that is a bit difficult, especially with this involved of a useEffect function

Reply:
I tested a version of this where I flattened the align prop and used that as the dependencies and it worked fine. There's probably a cleaner way to do this or break down the useEffect into standalone bits. But the general idea is object props like this in dependencies need to really be flattened so that they don't unnecessarily retrigger the function involved.
My hack:

```jsx
  const alignLeft = align?.left;
  const alignRight = align?.right;
  const alignTop = align?.top;
  const alignBottom = align?.bottom;
  ...
  // replace align.xxx in the function with the copy above
  ...
, [ alignLeft, alignRight, alignTop, alignBottom, ... ]
```